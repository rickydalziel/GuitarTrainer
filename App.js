import React, { Component } from 'react'
import { StyleSheet, Button, Text, View } from 'react-native'

export default class App extends Component {
  state = { clickCount: 0 }

  render () {
    const { clickCount } = this.state
    return (
      <View style={styles.container}>
        <Text style={styles.title}>Guitar Trainer</Text>
        <Text style={styles.instructions}>Please select a training option below</Text>
        <Text style={styles.instructions}>Button has been pressed {clickCount} times</Text>
        <Button title='Note Teacher' onPress={() => this.setState({ clickCount: clickCount + 1 })} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 10
  }
})
