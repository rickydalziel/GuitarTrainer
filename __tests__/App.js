import ReactNative from 'react-native'
import React from 'react'
import App from '../App'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  expect(renderer.create(<App />)).toMatchSnapshot()
})

it('increments a counter when the note teacher button is pressed', () => {
  const app = renderer.create(<App />).root
  const button = app.findByType(ReactNative.Button)

  button.props.onPress()

  const counter = app.findAllByType(ReactNative.Text)[2]
  expect(counter.props.children[1]).toBe(1)
})
