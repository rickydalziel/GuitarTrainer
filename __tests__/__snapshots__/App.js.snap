// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Guitar Trainer
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Please select a training option below
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Button has been pressed 
    0
     times
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Note Teacher
      </Text>
    </View>
  </View>
</View>
`;
